src/parquet/arrow/schema.cc

```diff
@@ -183,12 +184,31 @@ Status FromPrimitive(const PrimitiveNode* primitive, TypePtr* out) {
   return Status::OK();
 }
 
-Status StructFromGroup(const GroupNode* group, TypePtr* out) {
-  std::vector<std::shared_ptr<Field>> fields(group->field_count());
+// Forward declaration
+Status NodeToFieldInternal(const NodePtr& node,
+    const std::unordered_set<NodePtr>* needed_nodes, std::shared_ptr<Field>* out);
+
+bool IsNeededNode(const NodePtr& node, const std::unordered_set<NodePtr>* needed_nodes) {
```

Comment:
A needed node is a parquet schema leaf node which is required in the partial schema. 
So for example IsNeededNode is a shortcut used later to test if we want the parquet leaf node in the arrow schema so should maybe be an inline function or even a macro.
Maybe I should rename it to partial schema leaf node or something similar? 

About the tests - no problem, but I also want to test ReadTable, which needs a parquet file and not just the parquet schema. Is there a test suite that construct one?